### PR TITLE
SDK update

### PR DESCRIPTION
**Pull request by nekonomicon**
Applied some changes from git version of hlsdk(2013)
Applied many public bugfixes by @AlliedModders, @SamVanheer, @LevShisterov, @YaLTeR and others.
Added working spectator(allow_spectators cvar)
Added cl_viewbob cvar for viewmodel bobbing
Added mp_selfgauss cvar to turn on/off self gauss(same as in AG)
Added mp_satchelfix cvar for knowledge bugs with not deactivated satchels in HLDM
Added mp_bhopcap cvar for bunnyhop cap
Added cl_scoreboard_bg cvar for dark scoreboard background
Fixed menuselect cmd behavior
Fixed hud_takesshots cvar behavior
Scoreboard will show long nicknames(31 characters) now

### CODE REVIEW

File: common/const.h [hunk 1]
```diff
@@ -724,7 +728,8 @@ enum
 	kRenderFxDeadPlayer,		// kRenderAmt is the player index
 	kRenderFxExplode,			// Scale up really big!
 	kRenderFxGlowShell,			// Glowing Shell
-	kRenderFxClampMinScale		// Keep this sprite from getting very small (SPRITES only!)
+	kRenderFxClampMinScale,		// Keep this sprite from getting very small (SPRITES only!)
+	kRenderFxLightMultiplier	//CTM !!!CZERO added to tell the studiorender that the value in iuser2 is a lightmultiplier
```
Comment (a1batross) [inline common/const.h]: Where this comes from?

File: common/const.h [hunk 2]
```diff
@@ -110,7 +110,9 @@
 #define EF_NOINTERP			32	// don't interpolate the next frame
 #define EF_LIGHT			64	// rocket flare glow sprite
 #define EF_NODRAW			128	// don't draw entity
-
+#define EF_NIGHTVISION			256	// player nightvision
+#define EF_SNIPERLASER			512	// sniper laser effect
+#define EF_FIBERCAMERA			1024	// fiber camera
```
Comment (a1batross) [inline common/const.h]: Same here.

File: common/const.h [hunk 3]
```diff
@@ -621,8 +624,9 @@
 #define CHAN_BODY			4
 #define CHAN_STREAM			5	// allocate stream channel from the static or dynamic area
 #define CHAN_STATIC			6	// allocate channel from the static area 
-#define CHAN_NETWORKVOICE_BASE	7	// voice data coming across the network
+#define CHAN_NETWORKVOICE_BASE		7	// voice data coming across the network
 #define CHAN_NETWORKVOICE_END		500	// network voice data reserves slots (CHAN_NETWORKVOICE_BASE through CHAN_NETWORKVOICE_END).
+#define CHAN_BOT			501	// channel used for bot chatter.
```
Comment (a1batross) [inline common/const.h]: And there.

File: common/cvardef.h [hunk 4]
```diff
@@ -24,14 +24,15 @@
 #define FCVAR_SPONLY	(1<<6)	// This cvar cannot be changed by clients connected to a multiplayer server.
 #define FCVAR_PRINTABLEONLY	(1<<7)	// This cvar's string cannot contain unprintable characters ( e.g., used for player name etc ).
 #define FCVAR_UNLOGGED	(1<<8)	// If this is a FCVAR_SERVER, don't log changes to the log file / console if we are creating a log
+#define FCVAR_NOEXTRAWHITEPACE (1<<9)  // strip trailing/leading white space from this cvar
```
Comment (a1batross) [inline common/cvardef.h]: Is this flag is persist in GoldSrc? If so, should be added in Xash3D too.

File: pm_shared/pm_shared.h [hunk 5]
```diff
@@ -16,6 +16,12 @@
 #ifndef PM_SHARED_H
 #define PM_SHARED_H
 
+#ifdef _WIN32
+#ifndef __MINGW32__
+#pragma once
```
Comment (a1batross) [inline pm_shared/pm_shared.h]: MinGW supports this, AFAIK. At least, in newer versions.

File: common/const.h (hunk 1 above)
Comment (SamVanheer) [inline common/const.h]: These flags are all from the Github SDK: https://github.com/ValveSoftware/halflife/blob/5d761709a31ce1e71488f2668321de05f791b405/common/const.h#L714

File: pm_shared/pm_shared.h (hunk 5 above)
Comment (nekonomicon) [inline pm_shared/pm_shared.h]: Seems works with any compiler now...
Ok, will be removed.

File: common/const.h (hunk 1 above)
Comment (a1batross) [inline common/const.h]: Woah. I should update engine headers.

File: dlls/player.cpp [hunk 6]
```diff
@@ -2586,6 +2586,11 @@ void CBasePlayer::PostThink()
 
 	UpdatePlayerSound();
 pt_end:
+	if( pev->deadflag == DEAD_NO )
```
Comment (mittorn) [inline dlls/player.cpp]: вот блин. ломаете мне возможность поворачивать с кастомным клиентом. Ну ладно, хоть бегать пока могу.

Comment (nekonomicon) [reply dlls/player.cpp]: Что-то не так?

Comment (mittorn) [reply dlls/player.cpp]: Да всё норм :)

Comment (nekonomicon) [reply dlls/player.cpp]: Могу под квар убрать.

**Pull request merged**